appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -197,12 +210,32 @@ internal class ExperienceRenderer(
         stateMachines.getOwner(renderContext)?.stateMachine?.handleAction(StartStep(stepReference))
     }
 
-    suspend fun preview(experienceId: String): Boolean {
-        repository.getExperiencePreview(experienceId)?.let {
-            return show(it)
-        }
+    sealed class PreviewResponse {
+        object Success : PreviewResponse()
+        data class NetworkError(val throwable: Throwable?) : PreviewResponse()
+        data class HttpError(val code: Int?) : PreviewResponse()
+        data class PreviewDeferred(val experience: Experience, val frameId: String) : PreviewResponse()
+        data class StateMachineError(val experience: Experience, val error: Error) : PreviewResponse()
+    }
```

Comment:
Could we simplify to remove `PreviewResponse`? I'm thinking that `show()` could just return `ResultOf<Unit, RenderingError>` and a preview is just another form of rendering that propagates that error back up. The `RenderingError` could include relevant networking errors that are applicable to preview.

Reply:
Did not get. a good impl working here, the network errors seems wrong to mix with RenderingResult here